eclair-core/src/main/scala/fr/acinq/eclair/channel/publish/MempoolTxMonitor.scala

```diff
@@ -134,6 +133,7 @@ private class MempoolTxMonitor(nodeParams: NodeParams,
   def waitForConfirmation(): Behavior[Command] = {
     val messageAdapter = context.messageAdapter[CurrentBlockHeight](cbc => WrappedCurrentBlockHeight(cbc.blockHeight))
     context.system.eventStream ! EventStream.Subscribe(messageAdapter)
+    context.system.eventStream ! EventStream.Publish(TransactionPublished(txPublishContext.channelId_opt.getOrElse(ByteVector32.Zeroes), txPublishContext.remoteNodeId, cmd.tx, cmd.fee, cmd.desc))
```

Comment:
Doesn't this make us emit a `TransactionPublished` with each passing block?

Reply:
I don't think so, what makes you say that?
This code is only run when we explicitly call `waitForConfirmation()`, which we do in only one place.
Then we simply send ourselves a `CheckTxConfirmations` message at each block.

Reply:
I read to fast, thought we were in the `case WrappedCurrentBlockHeight(currentBlockHeight)` handler.